### ISSUE TITLE
link cli-ext with hasura cli

### ISSUE THREAD

**Issue by arvi3411301** [hasura/graphql-engine#4105]
Currently, `cli-ext` plugin manifest is placed under the plugins index repo which points to the latest version.

We need to link the `cli-ext` with the hasura cli version.

**Comment by arvi3411301** [comment]
@shahidhk We can fix this by two possible ways:

1. By embedding the `cli-ext` manifest in the CLI as an internal index. CLI will always check the internal index first for installing a plugin before falling back to the index repo.

2. By changing the structure of the `plugins-index` repo to include versions as subdirectories. For this, we need to rewrite the plugin's codebase to handle the versions.

@wawhal Please provide your feedback too.

**Comment by shahidhk** [comment]
I think we will not need multiple versions for plugins other than `cli-ext`. The reason or having multiple versions for `cli-ext` is to have a matching one for the CLI version since they both have a contract between them.

**Comment by picosam** [comment]
Hello, what's the current workaround to install this explicitly? I'm currently getting an error when trying to run the following command from Bitbucket pipelines:
```
+ hasura metadata apply --project hasura --endpoint $HASURA_ENDPOINT --admin-secret $HASURA_ADMIN_SECRET
{"level":"info","msg":"Applying metadata...","time":"2020-04-04T17:02:55Z"}
time="2020-04-04T17:02:57Z" level=fatal msg="failed to apply metadata: cannot apply metadata on the database: cannot build actions from metadata: error in converting sdl to metadata: fork/exec /root/.hasura/plugins/bin/hasura-cli_ext: no such file or directory: "
```

**Comment by hgiasac** [comment]
@picosam New CLI needs extra library `libstdc` that is missing in base alpine image. You try installing this before run hasura CLI

```
RUN apk add --no-cache libstdc++
```

**Comment by picosam** [comment]
Thanks @hgiasac 
Will this continue to be the case after the release of v1.2.0?

**Comment by picosam** [comment]
So, I added the library above, but I'm still getting the same error.

**Comment by hgiasac** [comment]
@picosam Sorry, it still needs more library. I can try one of them:

- Run another step, use another image
```yaml
...
- step:
    image: frolvlad/alpine-glibc
    script:
      - apk add curl libstdc++
      - curl -L# -f -o "/usr/local/bin/hasura" "https://github.com/hasura/graphql-engine/releases/download/v1.2.0-beta.3/cli-hasura-linux-amd64"
      - chmod +x "/usr/local/bin/hasura"
      - hasura migrate apply --project hasura --endpoint $HASURA_ENDPOINT --admin-secret $HASURA_ADMIN_SECRET
      - hasura metadata apply --project hasura --endpoint $HASURA_ENDPOINT --admin-secret $HASURA_ADMIN_SECRET
```
- Install directly in deploy script:
```yaml
image: atlassian/pipelines-awscli

definitions:
  step: &deploy
    services:
      - docker
    script:
      - docker build -t $AWS_ECR_REPO .
      - pipe: atlassian/aws-ecr-push-image:1.1.1
        variables:
          IMAGE_NAME: $AWS_ECR_REPO
          TAGS: "${BITBUCKET_COMMIT} latest"
      - aws ecs update-service --cluster $AWS_CLUSTER --service $AWS_SERVICE --force-new-deployment
      
      - wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub
      - wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.31-r0/glibc-2.31-r0.apk
      - wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.31-r0/glibc-bin-2.31-r0.apk
      - wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.31-r0/glibc-i18n-2.31-r0.apk
      - apk add glibc-2.31-r0.apk glibc-bin-2.31-r0.apk glibc-i18n-2.31-r0.apk
      - /usr/glibc-compat/bin/localedef -i en_US -f UTF-8 en_US.UTF-8
      - apk add curl libstdc++
      - curl -L# -f -o "/usr/local/bin/hasura" "https://github.com/hasura/graphql-engine/releases/download/v1.2.0-beta.3/cli-hasura-linux-amd64"
      - chmod +x "/usr/local/bin/hasura"
      - hasura migrate apply --project hasura --endpoint $HASURA_ENDPOINT --admin-secret $HASURA_ADMIN_SECRET
      - hasura metadata apply --project hasura --endpoint $HASURA_ENDPOINT --admin-secret $HASURA_ADMIN_SECRET
```

**Comment by RapidOwl** [comment]
I used @hgiasac suggestion above (frolvlad/alpine-glibc + installing libstdc++) and this solved the same CLI issue on GitHub Actions.

**Comment by jflambert** [comment]
Using `alpine-glibc` was not an option for me, so I ended up adapting @hgiasac's second script.

@tirumaraiselvan I spent quite a bit of time on this. I strongly suggest you produce an alpine-compatible `cli-ext` binary, or at least provide official instructions. Here's my `Dockerfile` for reference.

```
# use rabbitmq as a base image, since the apk is not available in alpine as of 3.12
FROM rabbitmq:3-alpine as rabbitmq

# Fetch additional tools and libraries
RUN apk --update add go postgresql-client curl gcc libc-dev libstdc++6 && \
    # install glibc for hasura-cli
    # https://github.com/hasura/graphql-engine/issues/4105#issuecomment-609639030
    wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub && \
    wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.32-r0/glibc-2.32-r0.apk && \
    wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.32-r0/glibc-bin-2.32-r0.apk && \
    wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.32-r0/glibc-i18n-2.32-r0.apk && \
    apk add glibc-2.32-r0.apk && \
    # handle glibc trigger error
    # https://github.com/sgerrand/alpine-pkg-glibc/issues/119#issuecomment-626627458
    rm -f /usr/glibc-compat/lib/ld-linux-x86-64.so.2 && \
    ln -s /usr/glibc-compat/lib/ld-2.31.so /usr/glibc-compat/lib/ld-linux-x86-64.so.2 && \
    apk add glibc-bin-2.32-r0.apk glibc-i18n-2.32-r0.apk && \
    /usr/glibc-compat/bin/localedef -i en_US -f UTF-8 en_US.UTF-8 && \
    # install hasura-cli
    curl -L# -f -o /bin/hasura-cli https://github.com/hasura/graphql-engine/releases/download/v1.3.0/cli-hasura-linux-amd64 && chmod +x /bin/hasura-cli &&\
    # test all tools
    go version && rabbitmq-diagnostics version && pg_isready -V && hasura-cli version --skip-update-check
```
